MathPortal/MathPortal/Source/Data/User.swift

```diff
@@ -78,10 +75,21 @@ class User: ParseObject {
     
     static func generateQueryWithUsername(_ username: String ) -> PFQuery<PFObject>? {
         let query = PFUser.query()
-        query?.whereKey(Object.username.rawValue, equalTo: username)
+        query?.whereKey(Object.username.rawValue, equalTo: username) 
         return query
     }
     
+    static func generateUserQuery() -> PFQuery<PFObject>? {
+        let query = PFUser.query()
+        return query
+    }
+    
+    static func fetchUserWithUserId(_ userId: String, completion: ((_ object: User?, _ error: Error?) -> Void)?)  {
+        generateUserQuery()?.getObjectInBackground(withId: userId, block: { (_ object: PFObject?, _ error: Error?) in
```

Comment:
This is a possible issue if `generateUserQuery` would return `nil`. In that case `completion` would not be called. Make sure in such cases to ALWAYS guard and call completion.